bugsnag-android-core/src/main/java/com/bugsnag/android/EventPayload.kt

```diff
@@ -13,25 +15,85 @@ import java.io.IOException
 class EventPayload @JvmOverloads internal constructor(
     var apiKey: String?,
     event: Event? = null,
-    internal val eventFile: File? = null,
+    eventFile: File? = null,
     notifier: Notifier,
     private val config: ImmutableConfig
-) : JsonStream.Streamable {
+) : JsonStream.Streamable, Deliverable {
 
-    var event = event
-        internal set(value) { field = value }
+    var event: Event? = event
+        internal set
+
+    internal var eventFile: File? = eventFile
+        private set
+
+    private var cachedBytes: ByteArray? = null
+
+    private val logger: Logger get() = config.logger
 
     internal val notifier = Notifier(notifier.name, notifier.version, notifier.url).apply {
         dependencies = notifier.dependencies.toMutableList()
     }
 
     internal fun getErrorTypes(): Set<ErrorType> {
         val event = this.event
-        return when {
-            event != null -> event.impl.getErrorTypesFromStackframes()
-            eventFile != null -> EventFilenameInfo.fromFile(eventFile, config).errorTypes
-            else -> emptySet()
+
+        return event?.impl?.getErrorTypesFromStackframes() ?: (
+            eventFile?.let { EventFilenameInfo.fromFile(it, config).errorTypes }
+                ?: emptySet()
+            )
+    }
+
+    private fun decodedEvent(): Event {
+        val localEvent = event
+        if (localEvent != null) {
+            return localEvent
         }
+
+        val eventSource = MarshalledEventSource(eventFile!!, apiKey ?: config.apiKey, logger)
+        val decodedEvent = eventSource()
+
+        // cache the decoded Event object
+        event = decodedEvent
```

Comment:
Maybe call this `cachedEvent`? Or is it already an exposed API?

Reply:
Unfortunately it's already exposed API. It's also sometimes the "original" event (when we don't have an event file)